frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx

```diff
@@ -48,13 +53,25 @@ const MetricsQueryInput: React.FC = () => {
   }, [dispatch, metric, namespace]);
 
   React.useEffect(() => {
-    const query = queries?.query;
+    const q = queries?.query;
     const text = queries?.text;
-    if (text && text.localeCompare(query) !== 0) {
+    if (text && text.localeCompare(q) !== 0) {
       setTitle(CUSTOM_QUERY);
       setIsPromQlDisabled(true);
+      if (query) {
+        removeQueryArgument('query0');
+      }
```

Comment:
what's the use of this , i see one in onChange

Reply:
It removes query params from the URL. If the user starts editing the promql or changes the query using the query dropdown.

Reply:
can we handle both scenarios in hook itself